spacy/tokens/doc.pyx

```diff
@@ -804,6 +804,14 @@ cdef class Doc:
             attr_ids[i] = attr_id
         if len(array.shape) == 1:
             array = array.reshape((array.size, 1))
+        # Check that all heads are within the document bounds
+        if HEAD in attrs:
+            col = attrs.index(HEAD)
+            for i in range(length):
+                # cast index to signed int
+                abs_head_index = numpy.int32(array[i, col]) + i
```

Comment:
Is there a reason we need to do this in numpy, rather than in C? The C equivalent would be `(<int32>array[i, col]) + 1` right?

I'm not really worried about performance here but the code just looks a little odd to me.

Reply:
No good reason, I'll change it.

Reply:
Wait, I remember that I did try this first. With `<int32_t>array[i, col]` I get errors:

```
OverflowError: Python int too large to convert to C long
```

Am I doing something obvious wrong here? I guess the cast in `Token.set_struct_attr()` is simpler because it's just C?

Reply:
Aaah right because it's a uint64. Yeah okay let's keep it like this.